src/Domain/Contracts/HasDetails.php

```diff
@@ -12,7 +12,7 @@ interface HasDetails extends Insight
     /**
      * Returns the details of the insight.
      *
-     * @return array<string>
+     * @return array<\NunoMaduro\PhpInsights\Domain\Details>
```

Comment:
This should be `@return array<int, \NunoMaduro\PhpInsights\Domain\Details>`

Reply:
Ups, this is a breaking change. Here we still need to support the string value : `@return array<int, \NunoMaduro\PhpInsights\Domain\Details|string>`.

Reply:
Hmm supporting that would require a lot of work for json formatter.
However is it actually a major breaking change as the class is marked internal?

Reply:
:thinking: If all implementations of HasDetails has been updated, it should be ok, no ?